server/routes/events.js

```diff
@@ -12,101 +13,498 @@ router.get('/ping', function(req, res, next) {
 
 // GET all events
 router.get('/', function(req, res, next) {
-  db.query('SELECT * FROM event', [], (err, result) => {
-    if (err) return next(err);
-    res.send(result.rows);
-  });
+  knex('event').select()
+  .then(result => {
+    if (result.length) {
+      res.json(result);
+    } else {
+      res.status(404).json('No events found.');
+    }
+  })
+  .catch(err => { return next(err) });
+});
+
+// Add a single event
+router.post('/', function(req, res, next) {
+  let values = { 
+  	fb_id: req.body.event_id, 
+  	name: req.body.name, 
+  	starts_at: req.body.starts_at, 
+  	ends_at: req.body.ends_at,
+  	quarter: req.body.quarter,
+  	location_id: req.body.location_id,
+  	description: req.body.description,
+  	fb_event: req.body.fb_event,
+  	picture: req.body.picture,
+  	is_featured: req.body.is_featured
+  };
+
+  Object.keys(values).forEach(function(key) {
+  	if (!values[key]) {
+  		//validReq = false;
+  		if (key == 'fb_id'){
+  			util.throwError(400, 'Missing \'event_id\' parameter.');
+  		}
+  		else{
+  			util.throwError(400, 'Missing \'' + key + '\' parameter.');
+  		}
+  	}
+	});
+
+  let category_ids = req.body.categories;
+  let category_values = [];
+  let category_ids_length;
+
+  if (category_ids){
+  	category_ids_length = category_ids.length;
+
+	  for (var i = 0; i < category_ids_length; i++){
+	  			category_values.push({ event_id: req.body.event_id, category_id: category_ids[i] });
+	  }
+  } 
+  else{
+  	util.throwError(400, 'Missing \'categories\' parameter.');
+  }
+
+  let company_ids = req.body.companies;
+	let company_values = [];
+  let company_ids_length;
+
+  if (company_ids){
+  	company_ids_length = company_ids.length;
+		for (var i = 0; i < company_ids_length; i++){
+	  			company_values.push({ event_id: req.body.event_id, company_id: company_ids[i] });
+	  }
+	}
+	else{
+		util.throwError(400, 'Missing \'companies\' parameter.');	
+	}
+
+  let host_ids = req.body.hosts;
```

Comment:
Since we have this so repetitively, just condense the code somewhat

e.g.

if (some param is not valid) { throwError}

let category_values = [];
let company_values = [];
let host_values = [];
bla.forEach(push to category_values);
bla.forEach(push to company_values);
bla.forEach(push to host_values);